tests/foreman/maintain/test_clone.py

```diff
@@ -0,0 +1,163 @@
+"""Tests for satellite-maintain clone
+
+:Requirement: foreman-maintain
+
+:CaseAutomation: Automated
+
+:CaseLevel: Component
+
+:CaseComponent: SatelliteClone
+
+:Assignee: lpramuk
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import pytest
+import requests
+
+from robottelo import constants
+from robottelo.config import settings
+
+USER = settings.subscription.rhn_username
+PASSWORD = settings.subscription.rhn_password
+SSH_PASS = settings.server.ssh_password
+POOL = settings.subscription.rhn_poolid
+BACKUP_DIR = '/var/backup'
+
+
+@pytest.mark.parametrize(
+    "sat_ready_rhel", [7, 8] if settings.server.version.rhel_version < 8 else [8], indirect=True
+)
+@pytest.mark.parametrize('backup_type', ['online', 'offline'])
+def test_positive_clone_backup(target_sat, sat_ready_rhel, backup_type):
+    rhel_version = sat_ready_rhel._v_major
+
+    # SATELLITE PART - SOURCE SERVER
+    # Enabling and starting services
+    assert target_sat.execute('satellite-maintain service enable').status == 0
+    assert target_sat.execute('satellite-maintain service start').status == 0
+    # Doing backup
+    backup_result = target_sat.execute(
+        f'satellite-maintain backup {backup_type} --assumeyes /var/backup'
+    )
+    assert backup_result.status == 0
+    # Getting backup directory
+    sat_backup_dir = backup_result.stdout.strip().split()[-2]
+
+    source_sat = target_sat
+    # Copying satellite backup to target RHEL
+    assert (
+        source_sat.execute(
+            f'sshpass -p "{SSH_PASS}" scp -o StrictHostKeyChecking=no -r {sat_backup_dir}'
+            f'root@{sat_ready_rhel.hostname}:/backup/'
+        ).status
+        == 0
+    )
+
+    # RHEL PART - TARGET SERVER
+    assert sat_ready_rhel.execute('ls /backup/config_files.tar.gz').status == 0
+    # Registering to cdn
+    sat_ready_rhel.register_to_cdn()
+    # Disabling repositories
+    assert sat_ready_rhel.execute('subscription-manager repos --disable=*').status == 0
+    # Getting satellite maintenance repo
+    res = requests.get(
+        f'http://ohsnap.sat.engineering.redhat.com/api/releases/{target_sat.version}/'
```

Comment:
Read this domain name from settings, shouldn't be hardcoded.

Reply:
There already is a suggestion by @Gauravtalreja1 mentioning the use of `sat_ready_rhel.download_repofile()`.